### PR TITLE
chore(ci): enable avx512 builds in the FFI CI

### PR DESCRIPTION
**Pull request by IceTDrinker**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by IceTDrinker** [comment]
We were discussing perf with @BourgerieQuentin and trying to enable avx512 on my laptop which does not have avx512 made no difference running the ffi, so enabling avx512 on x86 chips is safe thanks to some dynamic dispatch we do.

The upside is machines with avx512 will run concrete-core-ffi faster

File: .github/workflows/ffi_github_release.yml [hunk 1]
```diff
@@ -49,15 +54,20 @@ jobs:
...
+          toolchain: nightly
+          override: true
       - name: Build libconcrete_core_ffi
         env:
           RUSTFLAGS: "-Ctarget-feature=+aes,+sse2"
         run: |
           TMP_DIR="$(mktemp -d)"
 
           ./script/action_utils/concrete-core-ffi-release.sh \
-          --rust-toolchain "$(cargo xtask get_build_toolchain | tail -n 1)" \
```
Comment (agnesLeroy) [inline .github/workflows/ffi_github_release.yml:59]: Why did we get the toolchain this way before?

File: concrete-core-ffi/CMakeLists.txt [hunk 2]
```diff
@@ -1,8 +1,6 @@
 # concrete-core-ffi/CMakeLists.txt
 cmake_minimum_required(VERSION 3.16)
 project(concrete-core-ffi C)
-include_directories(${CONCRETE_BUILD_DIR})
-link_directories(${CONCRETE_BUILD_DIR})
```
Comment (agnesLeroy) [inline concrete-core-ffi/CMakeLists.txt:5]: Those lines were not used?

**Review by agnesLeroy — APPROVED** [review]
Thank you for this PR! I'm a bit confused because I thought we had activated this in the build some time ago but then reverted the change :thinking: but maybe it was something else.

File: concrete-core-ffi/CMakeLists.txt (hunk 2 above)
Comment (IceTDrinker) [reply concrete-core-ffi/CMakeLists.txt:5]: Legacy from very old ffi, removing those lines does not cause issues

File: .github/workflows/ffi_github_release.yml (hunk 1 above)
Comment (IceTDrinker) [reply .github/workflows/ffi_github_release.yml:59]: Basically to use avx512 we need the nightly toolchain, before we were using stable for x86 and nightly for aarch64, the C api is kind of an outlier now that all builds require a nightly toolchain

**Comment by IceTDrinker** [comment]
actually after checking the code some more it probably does not emit avx512 instructions, I'll check the binaries and the asm in there

**Comment by IceTDrinker** [comment]
so there is some avx512 code, it's not clear to me if it gets called however

**Pull request merged**